src/lib.rs

```diff
@@ -1303,7 +1307,62 @@ pub trait Drain {
         record: &Record,
         values: &OwnedKVList,
     ) -> result::Result<Self::Ok, Self::Err>;
-
+    /// Check if messages at the specified log level are enabled for this logger.
+    ///
+    /// Since disabled log levels are guaranteed to have their records ignored,
+    /// it's safe to skip logging messages at these level.
+    /// This can be used to avoid expensive computation of log message
+    /// arguments if the message would be ignored anyway.
+    ///
+    /// The logger is still free to ignore records even if the level is enabled,
+    /// an enabled level doesn't necessarily guarantee that the record will actually be logged.
+    /// ````
+    /// # #[macro_use]
+    /// # extern crate slog;
+    /// # use slog::*;
+    /// # fn main() {
+    /// let logger = Logger::root(Discard, o!());
+    /// if logger.is_enabled(Level::Debug) {
+    ///     let num = 5.0f64;
+    ///     let sqrt = num.sqrt();
+    ///     debug!(logger, "Sqrt"; "num" => num, "sqrt" => sqrt);
+    /// }
+    /// # }
+    /// ````
+    #[inline]
+    fn is_enabled(&self, level: Level) -> bool {
+        level.as_usize() <= ::__slog_static_max_level().as_usize()
+    }
+    /// Check if messages at the critical log level are enabled for this logger via `is_enabled`.
+    #[inline]
+    fn critical_enabled(&self) -> bool {
```

Comment:
Wouldn't `is_x_enabled()` be more idiomatic?

Reply:
Fixed in the amended commit